client/lcd/root.go

```diff
@@ -127,18 +127,20 @@ func (rs *RestServer) Start(listenAddr string, sslHosts string,
 		if err != nil {
 			return
 		}
-		go rpcserver.StartHTTPAndTLSServer(
+
+		rs.log.Info("Starting Gaia Lite REST service...")
+		rs.log.Info(rs.fingerprint)
+
+		if err := rpcserver.StartHTTPAndTLSServer(
 			rs.listener,
 			rs.Mux,
 			certFile, keyFile,
 			rs.log,
-		)
-		rs.log.Info(rs.fingerprint)
-		rs.log.Info("REST server started")
+		); err != nil {
+			return err
+		}
```

Comment:
lol this is a bit nutty :P, let's separate out `err` variable definition from the if statement 

```
err := ....
if err != nil {
   ... 
}
```

Reply:
As you wish. Updated.